src/app/classes/classes.component.ts

```diff
@@ -9,19 +9,47 @@ import { SubscriptionService } from '../services/subscription.service';
   styleUrls: ['./classes.component.css'],
 })
 export class ClassesComponent implements OnInit {
-  classes: Class[];
-  selectedClass: Class;
+  allClasses: Class[];
+  searchText = '';
+  displayedClasses: Class[];
+  subscribedClases: string[];
 
   constructor(private subService: SubscriptionService) {}
 
   ngOnInit() {
+    this.subService.getClasses().subscribe((classes) => {
+      (this.allClasses = classes), (this.displayedClasses = this.allClasses);
+    });
     this.subService
-      .getClasses()
-      .subscribe((classes) => (this.classes = classes));
+      .getSubscriptions()
+      .subscribe((subs) => (this.subscribedClases = subs));
   }
 
   onSelect(myclass: Class): void {
-    this.selectedClass = myclass;
-    this.subService.addSubscription(myclass.id);
+    this.subService
+      .addSubscription(myclass.id)
+      .then(() => {
+        console.log('Class added');
+      })
+      .catch((error) => {
+        console.log(error);
+      });
+  }
+
+  classFilter(): void {
+    //Clear serchable class list
+    this.displayedClasses = [];
+
+    //Only compare the class name and number
+    let name: string;
+    for (let item of this.allClasses) {
+      name = item.subject + ' ' + item.course;
+      if (
+        name.toLocaleLowerCase().includes(this.searchText) ||
```

Comment:
do we need to do both the localeLowerCase and direct includes?

Reply:
Yes, in the case a user types 'ece' instead of 'ECE', either case will work. If we don't check for both, search will only work for the exact capitalization match, in this case only 'ECE'.